module-workers.tf

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  worker_count_expected = coalesce(one(module.workers[*].worker_count_expected), 0)
+
+  # Distinct list of compartments for enabled worker pools
+  worker_compartments = distinct(compact([
+    for k, v in var.worker_pools : lookup(v, "compartment_id", local.compartment_id)
```

Comment:
If I'm reading this right, you are first looking for the `compartment_id` in each worker pool. If it's not present, try a general `worker_compartment_id`, which if not present try `compartment_id`, which if not present, try `compartment_ocid`. The last case I assume is catch for resource manager.

Reply:
That's correct, and will be documented with worker pools. I think we can drop the last fallback if #172 is completed.